tests/tracer/test_propagation.py

```diff
@@ -238,25 +258,44 @@ def test_extract_128bit_trace_ids():
 
     tracer = DummyTracer()
 
-    trace_id = int(os.getenv("TEST_TRACE_ID"))
-    trace_id_64bit = trace_id & 2 ** 64 - 1
-    # Get the hex representation of the 64 most signicant bits
-    trace_id_hob_hex = "{:032x}".format(trace_id)[:16]
-
-    headers = {
-        "x-datadog-trace-id": str(trace_id_64bit),
-        "x-datadog-tags": "=".join([HIGHER_ORDER_TRACE_ID_BITS, trace_id_hob_hex]),
-    }
-
-    context = HTTPPropagator.extract(headers)
-
-    tracer.context_provider.activate(context)
+    for trace_id in [2 ** 128 - 1, 2 ** 127 + 1, 2 ** 65 - 1, 2 ** 64 + 1, 2 ** 127 + 2 ** 63]:
+        trace_id_hex = "{:032x}".format(trace_id)
+        span_id = 1
+        span_id_hex = "{:016x}".format(span_id)
+
+        style = os.getenv("DD_TRACE_PROPAGATION_STYLE")
+        if style == "datadog":
+            # Get the hex representation of the 64 most signicant bits
+            trace_id_64bit = trace_id & 2 ** 64 - 1
+            headers = {
+                "x-datadog-trace-id": str(trace_id_64bit),
+                "x-datadog-parent-id": str(span_id),
+                "x-datadog-tags": "=".join([HIGHER_ORDER_TRACE_ID_BITS, trace_id_hex[:16]]),
+            }
+        elif style == "b3multi":
```

Comment:
rather than all of the if elifs to check the style and assert certain headers, could we parametrize the expected values for each header style, like how we do in the other tests? Or does that not work here?

Reply:
Same comment as above